kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/DataServiceOptions.java

```diff
@@ -155,6 +155,6 @@ int getRecoveryMaximumAllowedFailures() {
     }
 
     int getCriticalComponentTimeout() {
-        return getConnectDelay() * CONNECT_CRITICAL_COMPONENT_TIMEOUT_MULTIPLIER;
+        return getBackoffMaxTentatives() * CONNECT_CRITICAL_COMPONENT_TIMEOUT_MULTIPLIER;
```

Comment:
The timeout for the critical components was computed based on the connection delay. Now that the connection delay is replaced with the maximum number of attempts, a new formula should be used.